cradle-core/src/main/java/com/exactpro/cradle/CradleStorage.java

```diff
@@ -953,11 +957,12 @@ public final Collection<String> getScopes(BookId bookId) throws IOException, Cra
 	 * @throws CradleStorageException if given book ID is invalid
 	 */
 	public CompletableFuture<CradleResultSet<CounterSample>> getMessageCountersAsync(BookId bookId,
+																			   String page,
```

Comment:
Page can be resolved by Interval that is provided by client,
This interval can span over multiple pages.

Client facing API methods should not require passing page names, instead method should detect required pages and aggregate results.

This principle should be kept for other API methods as well